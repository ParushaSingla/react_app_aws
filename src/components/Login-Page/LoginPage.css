.form-style  {
    margin-top: 115px;
    border: solid;
    height: 300px
  }
  .label-style{
    margin: 30px 190px 0px;
  }
.button-style{
    margin: 30px 350px 0px;
}
