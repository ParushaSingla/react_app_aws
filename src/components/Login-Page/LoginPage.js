import React, { useContext, useState } from "react";
import "./LoginPage.css";
import { UserContext } from "../../contextAPI/userContext";
import { useToasts } from "react-toast-notifications";
//valid user details are kept hard coded
function userDetails(){
  return {
    username:"parusha",
    password:"singla"
  }
}
function LoginPage(props) {
  const context = useContext(UserContext);
  const [state, setState] = useState({
    username: "",
    password: "",
  });
  const { addToast } = useToasts();
  const inputFieldStyle = {
    marginLeft: "190px",
    width: "50%",
  };
  const validUserData = {
    username: "Parusha",
    isLoggedIn: true,
  };
  const handleChange = (e) => {
    const { id, value } = e.target;
    setState((prevState) => ({
      ...prevState,
      [id]: value,
    }));
  };

  const validateUser = (event) => {
    event.preventDefault();
    var appearanceValue = "";
    var toastMessage = "";
    const validUser=userDetails();
    if (state.username === validUser.username && state.password === validUser.password) {
      context.updateUserValue(validUserData);
      localStorage.setItem("loggedUser", JSON.stringify(validUserData));
      appearanceValue = "success";
      toastMessage = "Valid User!! logged in successfully";
      props.history.push("");
    } else {
      toastMessage = "UserName or Password is not correct!!!";
      appearanceValue = "error";
    }
    addToast(toastMessage, {
      appearance: appearanceValue,
      autoDismiss: true,
    });
  };
  return (
    <div className="container">
      <form className="form-style">
        <label className="label-style" htmlFor="exampleInputEmail1">
          Username
        </label>
        <input
          type="text"
          className="form-control"
          id="username"
          style={inputFieldStyle}
          aria-describedby="emailHelp"
          value={state.username}
          onChange={handleChange}
        />
        <label className="label-style" htmlFor="exampleInputPassword1">
          Password
        </label>
        <input
          type="password"
          className="form-control"
          style={inputFieldStyle}
          id="password"
          value={state.password}
          onChange={handleChange}
        />

        <button
          type="submit"
          className="btn btn-primary button-style"
          onClick={validateUser}
          disabled={state.username === "" ? true : state.password === ""}
          id="login-button"
        >
          Submit
        </button>
      </form>
    </div>
  );
}

export default LoginPage;
