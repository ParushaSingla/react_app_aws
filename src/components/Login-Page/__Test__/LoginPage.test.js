import React from "react";
import LoginPage from "../LoginPage";
import { ToastProvider } from "react-toast-notifications";
import { mount, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });
jest.mock("rc-util/lib/Portal");
let container;
beforeEach(() => {
  container = mount(
    <ToastProvider>
      <LoginPage />
    </ToastProvider>
  );
});

afterEach(() => {
  container.unmount();
  container = null;
});
test("Snapshot testing", () => {
  expect(container).toMatchSnapshot();
});
test("if fields are emplty the login button is disabled", () => {
  const stateSetter = jest.fn();
  jest.spyOn(React, "useState").mockImplementation((stateValue) => [
    (stateValue = {
      username: "",
      password: "",
    }),
    stateSetter,
  ]);
  container.setState({
    username: "",
    password: "",
  });
  expect(container.find("button").prop("disabled")).toBeTruthy();
});
