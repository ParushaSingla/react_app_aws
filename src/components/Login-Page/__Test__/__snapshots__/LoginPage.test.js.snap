// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot testing 1`] = `ReactWrapper {}`;

exports[`with ReactDOM 1`] = `ReactWrapper {}`;

exports[`with ReactDOM 2`] = `ReactWrapper {}`;
